platform.md

```diff
@@ -148,6 +155,439 @@ A platform MAY support any number of mixins for a given stack in order to suppor
 Changes introduced by mixins SHOULD be restricted to the addition of operating system software packages that are regularly patched with strictly backwards-compatible security fixes.
 However, mixins MAY consist of any changes that follow the [Compatibility Guarantees](#compatibility-guarantees).
 
+## Lifecycle Interface
+### Platform API Compatibility
+
+The platform SHOULD set `CNB_PLATFORM_API=<platform API version>` in the lifecycle's execution environment
+
+IF `CNB_PLATFORM_API` is set in the lifecycle's execution environment, the lifecycle:
+  - MUST EITHER conform to the matching version of this specification OR
+  - Fail if it does not support `<platform API version>`
+
+### Operations
+
+#### Build
+A single app image build* consists of the following phases:
+
+1. Detection
+1. Analysis
+1. Cache Restoration
+1. Build*
+1. Export
+
+A platform MUST execute these phases EITHER by invoking the following phase-specific lifecycle binaries in order:
+1. `/cnb/lifecycle/detector`
+1. `/cnb/lifecycle/analyzer`
+1. `/cnb/lifecycle/restorer`
+1. `/cnb/lifecycle/builder`
+1. `/cnb/lifecycle/exporter`
+
+OR by executing `/cnb/lifecycle/creator`.
+
+> \* **build** is an overloaded term that refers to both a single phase and the operation comprised of the above phases.
+> The meaning of any particular instance of the word **build** must be assessed in context
+
+#### Rebase
+When a new stack version with the same stack ID is available, an updated app image SHOULD be generated from the existing app image config by replacing the run image layers in the existing `RootFS` with the layers from the new run image.
+This is referred to as rebasing the app, launch, and launcher layers onto the new run image layers.
+When layers are rebased, any app image metadata referenceing to the original run image MUST be updated to reference to the new run image.
+This entire operation is referred to as rebasing the app image.
+
+Rebasing allows for fast stack updates for app images without requiring a rebuild. A rebase requires minimal data transfer when the app and run images are colocated on a Docker registry.
+
+To rebase an app image a platform MUST execute the `/cnb/lifecycle/rebaser` OR perform an equivalent operation.
+ 
+#### Launch
+`/cnb/lifecycle/launcher` is responsible for launching user and buildpack provided processes in the correct execution environment.
+`/cnb/lifecycle/launcher` SHALL be the `ENTRYPOINT` for all app images.
+
+### Usage
+
+All lifecycle phases:
+
+- MUST read `CNB_PLATFORM_API` from the execution environment and evaluate compatibility before attempting to parse other inputs (see [Platform API Compatibility](#platform-api-compatibility))
+- MUST give command line inputs precedence over other inputs
+
+#### `detector`
+The platform MUST execute `detector` in the build environment
```

Comment:
is it worth bolding the build environment in these sections to make it more visible?